lib/_007/Q.pm

```diff
@@ -129,10 +129,10 @@ class Q::Identifier does Q::Term {
 }
 
 class Q::Term::Regex does Q::Term {
-    has Q::Literal::Str $.contents;
+    has Val::Str $.contents;
```

Comment:
I'm having trouble putting into words why, but I think it should be `Q::Literal::Str` there, like you had it. Let me try to express it anyway.

* `Q` is about program structure. The regex structurally contains a sequence of atoms, some of which are literal strings. They are not string *values* (yet), they are string *literals* in the source code.
* Compare `Q::Term::Array`, which in the end is full of `Q::Expr`; or `Q::Term::Object` which is full of `Q::Property`.
* Admittedly, there's no iron-clad rule there. For example, why is the `$.key` of a `Q::Property` a `Val::Str` and not a `Q::Literal::Str`? Because there are many ways to write that key, only one of which is an actual string literal. But in the case of the `Q::Term::Regex`, it's always a string literal.
* Maybe the rule is that `Val::` objects are used "as late as possible" in `Q` structures. This rule is confounded a little bit because container `Q` types often use `Val::Array` (and occasionally `Val::Object`, soon-to-be-known-as `Val::Hash`) just to hold multiple values. But apart from that, it holds. `Val::Str` is only used sparingly in three places: `Q::Literal::Str` (obvs), `Q::Identifier`, and the aforementioned `Q::Property`.

This was my biggest objection to this patch. I'm not going to let this one block the merge, however, because I believe the above will prove correct on its own once we start adding in other regex features.